test/builder-api/src/channels.js

```diff
@@ -190,6 +212,17 @@ describe('Channels API', function () {
         });
     });
 
+    it('returns success on HEAD with the specified name, version, and release', function (done) {
+      request.head('/depot/channels/neurosis/foo/pkgs/testapp/0.1.3/20171205003213')
```

Comment:
i've added these checks above for the version check as opposed to the release since the release is likely a bit more specific and the routes have the same behaviors,

Reply:
@eeyun In my local testing the `target` query was only working as expected on the fully qualified package identifier. On the package and version endpoints, it would ALWAYS return 200, even if that package target did not exist.  That's why I suggested a negation test, checking for a 404. 

The only place in hab cli that [check_package](https://github.com/habitat-sh/habitat/blob/master/components/hab/src/command/pkg/upload.rs#L69) is used, it will be a fully qualified ident, with the release.

Reply:
For example:

```
[10][default:/src:0]# export HAB_ORIGIN=hank
[11][default:/src:0]# export HAB_AUTH_TOKEN=hank
[12][default:/src:0]# curl -XGET -H "Authorization: Bearer $HAB_AUTH_TOKEN" http://localhost/v1/depot/channels/hank/unstable/pkgs/testapp3/0.1.0/20200320152015?target=x86_64-windows
[13][default:/src:0]# curl -I -XGET -H "Authorization: Bearer $HAB_AUTH_TOKEN" http://localhost/v1/depot/channels/hank/unstable/pkgs/testapp3/0.1.0/20200320152015?target=x86_64-windows
HTTP/1.1 404 Not Found
Server: nginx/1.17.8
Date: Mon, 23 Mar 2020 14:06:37 GMT
Content-Length: 0
Connection: keep-alive

[14][default:/src:0]# hab pkg upload -u http://localhost /src/results/hank-testapp3-0.1.0-20200320152015-x86_64-linux.hart
    75 B / 75 B | [==============================================================================================================================================================================] 100.00 % 89.32 KB/s
→ Using existing public origin key hank-20200320151753.pub
→ Using existing hank/testapp3/0.1.0/20200320152015
[15][default:/src:0]# curl -I -XGET -H "Authorization: Bearer $HAB_AUTH_TOKEN" http://localhost/v1/depot/channels/hank/unstable/pkgs/testapp3/0.1.0/20200320152015?target=x86_64-windows
HTTP/1.1 404 Not Found
Server: nginx/1.17.8
Date: Mon, 23 Mar 2020 14:06:58 GMT
Content-Length: 0
Connection: keep-alive

[16][default:/src:0]# curl -I -XGET -H "Authorization: Bearer $HAB_AUTH_TOKEN" http://localhost/v1/depot/channels/hank/unstable/pkgs/testapp3/0.1.0/20200320152015?target=x86_64-linux
HTTP/1.1 200 OK
Server: nginx/1.17.8
Date: Mon, 23 Mar 2020 14:07:02 GMT
Content-Type: application/json
Content-Length: 1765
Connection: keep-alive
cache-control: private, no-cache, no-store
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Alternate-Protocol: 443:npn-spdy/3
X-UA-Compatible: IE=Edge
Strict-Transport-Security: max-age=31536000; includeSubDomains;

[17][default:/src:0]# curl -I -XGET -H "Authorization: Bearer $HAB_AUTH_TOKEN" http://localhost/v1/depot/channels/hank/unstable/pkgs/testapp3/0.1.0?target=x86_64-linux
HTTP/1.1 200 OK
Server: nginx/1.17.8
Date: Mon, 23 Mar 2020 14:07:11 GMT
Content-Type: application/json
Content-Length: 137
Connection: keep-alive
cache-control: private, no-cache, no-store
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Alternate-Protocol: 443:npn-spdy/3
X-UA-Compatible: IE=Edge
Strict-Transport-Security: max-age=31536000; includeSubDomains;

[18][default:/src:0]# curl -I -XGET -H "Authorization: Bearer $HAB_AUTH_TOKEN" http://localhost/v1/depot/channels/hank/unstable/pkgs/testapp3/0.1.0?target=x86_64-windows
HTTP/1.1 200 OK
Server: nginx/1.17.8
Date: Mon, 23 Mar 2020 14:07:23 GMT
Content-Type: application/json
Content-Length: 137
Connection: keep-alive
cache-control: private, no-cache, no-store
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Alternate-Protocol: 443:npn-spdy/3
X-UA-Compatible: IE=Edge
Strict-Transport-Security: max-age=31536000; includeSubDomains;

[19][default:/src:0]# curl -I -XHEAD -H "Authorization: Bearer $HAB_AUTH_TOKEN" http://localhost/v1/depot/channels/hank/unstable/pkgs/testapp3/0.1.0?target=x86_64-windows
HTTP/1.1 200 OK
Server: nginx/1.17.8
Date: Mon, 23 Mar 2020 14:07:34 GMT
Content-Type: application/json
Content-Length: 0
Connection: keep-alive
cache-control: private, no-cache, no-store
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Alternate-Protocol: 443:npn-spdy/3
X-UA-Compatible: IE=Edge
Strict-Transport-Security: max-age=31536000; includeSubDomains;

[20][default:/src:0]# curl -I -XHEAD -H "Authorization: Bearer $HAB_AUTH_TOKEN" http://localhost/v1/depot/channels/hank/unstable/pkgs/testapp3/0.1.0/20200320152015?target=x86_64-windows
HTTP/1.1 404 Not Found
Server: nginx/1.17.8
Date: Mon, 23 Mar 2020 14:07:45 GMT
Content-Length: 0
Connection: keep-alive

[21][default:/src:0]#
```

The only response that seems to be behaving as expected, when passed the target, is the fully qualified release - which happens to be the only one currently in use by the hab client when it checks on an `upload`. 🤔